R/utilities-sensitivity-analysis.R

```diff
@@ -1,342 +1,343 @@
+#' @param pkParameterResultsFilePath path to pk parameter results CSV
+#' @param resultsFileFolder folder where results of sensitivity analysis will be saved
+#' @param popSAResultsIndexFile Names of CSV file containing index of sensitivity analysis CSV files
+#' @param resultsFileName root name of sensitivity analysis results CSV files
+#' @param quantileVec vector of quantiles in (0,1).  For each output and pk parameter, there will be a distribution of pk parameter values for the population.
+#' The individuals yielding pk parameters closest to these quantiles will be selected for sensitivity analysis.  Default is (0.05,0.5,0.95)
+#' @param numberOfCores the number of cores to be used for parallelization of the sensitivity analysis.  Default is 1 core (no parallelization).
+#' @export
+populationSensitivityAnalysis <- function(simFilePath,
+                                          popDataFilePath,
+                                          pkParameterResultsFilePath,
+                                          resultsFileFolder,
+                                          resultsFileName,
+                                          popSAResultsIndexFile = "sensitivityAnalysesResultsIndexFile",
+                                          quantileVec = c(0.05, 0.5, 0.95),
+                                          numberOfCores = 1) {
+  sensitivityAnalysesResultsIndexFileDF <- getSAFileIndex(pkParameterResultsFilePath, quantileVec, resultsFileFolder, resultsFileName, popSAResultsIndexFile)
+  ids <- unique(sensitivityAnalysesResultsIndexFileDF$IndividualId)
+
+  analyzeSensitivity(simFilePath,
+    parametersToPerturb = NULL,
+    popFilePath = popDataFilePath,
+    individualID = ids,
+    numberOfCores = numberOfCores,
+    resultsFileFolder = resultsFileFolder,
+    resultsFileName = resultsFileName
+  )
+}
+
+
+#' @title getSAFileIndex
+#' @description Function to build and write to CSV a dataframe that stores all sensitivity analysis result files that will be output by a population sensitivity analysis.
+#' @param pkParameterResultsFilePath path to pk parameter results CSV file
+#' @param quantileVec quantiles of distributions of pk parameter results for a population.  The individual in the population that yields a pk parameter closest to the quantile is selected for sensitivity analysis.
+#' @param resultsFileFolder path to population sensitivity analysis results CSV files
+#' @param resultsFileName root name of population sensitivity analysis results CSV files
+#' @param popSAResultsIndexFile name of CSV file that will store index that identifies name of sensitivity analysis results file for each sensitivity analysis run
+
+getSAFileIndex <- function(pkParameterResultsFilePath,
+                           quantileVec,
+                           resultsFileFolder,
+                           resultsFileName,
+                           popSAResultsIndexFile) {
+
+  allPKResultsDataframe <- getPKResultsDataFrame(pkParameterResultsFilePath)
```

Comment:
I am not sure once again but I am wondering if we need the data frame at all. We have an object (PKParameterResults) or something like that that can return all values for a given output, pk parameter etc... I feel that we are constructing a data frame based on the PKResults just to get the same behavior. Of course, I may miss the point completely but I just wanted to raise the point :)

Reply:
The dataframe format was convenient for the later part of this function